plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergStatistics.java

```diff
@@ -862,6 +863,117 @@ public void testNoScalarColumns()
+        // create several snapshots
+        assertUpdate(writeSession, "INSERT INTO show_stats_after_expiration VALUES 1", 1);
+        assertUpdate(writeSession, "INSERT INTO show_stats_after_expiration VALUES 2", 1);
+        assertUpdate(writeSession, "INSERT INTO show_stats_after_expiration VALUES 3", 1);
+
+        long beforeAnalyzedSnapshot = getCurrentSnapshotId("show_stats_after_expiration");
+
+        assertUpdate(
+                Session.builder(getSession())
+                        .setCatalogSessionProperty(catalog, EXPIRE_SNAPSHOTS_MIN_RETENTION, "0s")
+                        .build(),
+                "ALTER TABLE show_stats_after_expiration EXECUTE expire_snapshots(retention_threshold => '0d')");
+        assertThat(query("SELECT count(*) FROM \"show_stats_after_expiration$snapshots\""))
+                .matches("VALUES BIGINT '1'");
+
+        assertUpdate(writeSession, "INSERT INTO show_stats_after_expiration VALUES 4", 1);
+        assertUpdate("ANALYZE show_stats_after_expiration");
+        long analyzedSnapshot = getCurrentSnapshotId("show_stats_after_expiration");
+
+        assertUpdate(writeSession, "INSERT INTO show_stats_after_expiration VALUES 5", 1);
+        long laterSnapshot = getCurrentSnapshotId("show_stats_after_expiration");
+
+        assertQuery(
+                "SHOW STATS FOR (SELECT * FROM show_stats_after_expiration FOR VERSION AS OF " + beforeAnalyzedSnapshot + ")",
+                """
+                        VALUES
+                          ('key', null, null, 0, null, '1', '3'), -- NDV not present, as ANALYZE was run on a later snapshot
+                          (null,  null, null, null, 3, null, null)""");
+
+        assertQuery(
+                "SHOW STATS FOR (SELECT * FROM show_stats_after_expiration FOR VERSION AS OF " + analyzedSnapshot + ")",
+                """
+                        VALUES
+                          ('key', null, 4, 0, null, '1', '4'), -- NDV present, this is the snapshot ANALYZE was run for
+                          (null,  null, null, null, 4, null, null)""");
+
+        assertQuery(
+                "SHOW STATS FOR (SELECT * FROM show_stats_after_expiration FOR VERSION AS OF " + laterSnapshot + ")",
+                """
+                        VALUES
+                          ('key', null, 4, 0, null, '1', '5'), -- NDV present, stats "inherited" from previous snapshot
+                          (null,  null, null, null, 5, null, null)""");
+
+        // Same as laterSnapshot but implicitly
+        assertQuery(
+                "SHOW STATS FOR show_stats_after_expiration",
+                """
+                        VALUES
+                          ('key', null, 4, 0, null, '1', '5'), -- NDV present, stats "inherited" from previous snapshot
+                          (null,  null, null, null, 5, null, null)""");
+
+        // Re-analyzing after snapshot expired
```

Comment:
Which "snapshot" are you referring to here?

Reply:
some snapshot expired